src/screens/ShowDetailsScreen.tsx

```diff
@@ -132,17 +142,16 @@ const ShowDetailsScreen: React.FC = () => {
 
     useEffect(() => {
         const handler = async () => {
-            setLoading(true);
```

Comment:
Curious about this change. Does this impact the loading behavior?

Reply:
I'm under the impression that this is a redundant `setStateAction` because `loading` is initialized to `true`. 

With this change, the loader functions as expected, as it did before.

Reply:
If `location` updates this useEffect will re-trigger. I think in that case we would want to display loading again before re-rendering the screen.

Reply:
if `location` updates, isn't the component (screen) re-rendered anyway, re-initializing `loading` to true?

Not sure I actually understand the use of `[location]` in the useEffect.

Given the route `/details/tv/1399` for example, is there ever an instance that `location` would update at the same route `/details/tv/1399`?

@Thenlie

Reply:
https://api.reactrouter.com/v7/functions/react_router.useLocation.html
I think this is at least the recommended approach. Not sure how much of an edge case it really is. But if only the `location` changes the entire screen will not re-render. I am not sure if there is ever a case where this happens, but it seems like react-router believes it can/does happen.

Reply:
Using `location` in a useEffect in a SPA structure makes perfect sense and I can see several cases where that would be practical to do. In Streamability's case, given that _any_ other route renders a different component (screen), I'm having a hard time understanding this.

I believe it to be impossible that `ShowDetailsScreen`'s `location` could 'update' while remaining at the same route. 
If `ShowDetailsScreen`'s route does 'update', that's a different route with different data.

If you navigate between `/details/tv/1399` and `/details/tv/1396` for example, I think this would be considered as a _new_ route, not an update to an existing one. Therefore, the entire screen is re-rendered, re-initializing `loading` to true.

Right!?!?!